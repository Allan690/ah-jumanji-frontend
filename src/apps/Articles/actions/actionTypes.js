//POSTING
export const POSTING_ARTICLE="article is posting";
export const POSTED_ARTICLE="article is posted";
export const ERROR_POSTING_ARTICLE="posting error";

//GETTING
export const GOT_ARTICLE="article is fetched";
export const ERROR_GETTING_ARTICLE="fetch error";

//UPDATING
export const UPDATING_ARTICLE="article is updating";
export const UPDATED_ARTICLE="article is updated";
export const UPDATING_CURRENT="Adding updates to the data";
export const UPDATED_ERROR="errror updating article";

//DELETING
export const DELETING_ARTICLE="article is being deleted";
export const DELETED_ARTICLE="article is deleted";
export const ERROR_DELETING_ARTICLE="error on article deleting";
