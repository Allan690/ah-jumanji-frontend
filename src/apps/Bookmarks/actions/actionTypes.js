// Actions types

export const BOOKMARK_SUCCESS="BOOKMARK_SUCCESS";
export const BOOKMARK_FAILED="BOOKMARK_FAILED";
