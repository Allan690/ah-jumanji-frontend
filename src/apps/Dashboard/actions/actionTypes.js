// Actions types

export const FETCH_SUCCESS="FETCH_SUCCESS";
export const FETCH_FAILED="FETCH_FAILED";
export const FETCH_COUNT="FETCH_COUNT";
