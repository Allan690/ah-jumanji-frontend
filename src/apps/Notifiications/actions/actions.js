export const GET_READ_NOTIFICATIONS = "GET_READ_NOTIFICATIONS";
export const GET_UNREAD_NOTIFICATIONS = "GET_UNREAD_NOTIFICATIONS";
export const GET_SPECIFIC_NOTIFICATION = "GET_SPECIFIC_NOTIFICATION";
export const MARK_AS_READ = "MARK_AS_READ";
export const MARK_AS_UNREAD = "MARK_AS_UNREAD";
export const FORM_SUBMIT = "SUBMIT_FORM_DATA";
export const FETCH_NOTIFICATIONS_SUCCESSFUL = "FETCH_NOTIFICATIONS_SUCCESSFUL";
export const FETCH_NOTIFICATIONS_FAILED = "FETCH_NOTIFICATIONS_FAILED";
