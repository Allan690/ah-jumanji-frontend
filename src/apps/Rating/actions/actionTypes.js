export const ADD_RATING_SUCCESS = "ADD_RATING_SUCCESS";
export const ADD_RATING_FAILED = "ADD_RATING_FAILED";
export const FETCH_AVERAGE_RATING = "FETCH_AVERAGE_RATING";
