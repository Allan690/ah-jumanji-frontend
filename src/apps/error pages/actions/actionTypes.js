export const NEW_ERROR_RAISED = "NEW_ERROR_RAISED";
export const CLEAR_ERROR_MESSAGE = "CLEAR_ERROR_MESSAGE";
