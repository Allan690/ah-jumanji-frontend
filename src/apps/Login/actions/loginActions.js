import axios from 'axios';

import * as types from  './actionTypes';

const endpoint ='https://ah-jumanji-staging.herokuapp.com/api/users/login';

export function LoginSucess(user){
    return {
        type: types.LOGIN_SUCCESS,
        user
    };
}

export function LoginFailed(error_message){
    return {
        type: types.LOGIN_FAILED,
        error_message
    };
}

export function authenticateUser(userData) {
    return function (dispatch) {
        return axios.post(endpoint,
        userData, {
          headers: {
            Accept: 'application/json'
          }
        })
        .then(res => {
            const token = res.data.user.token;
            localStorage.setItem('token', token);
            dispatch(LoginSucess(res.data));
            setTimeout(function(){
              window.location.replace('/home');
            }, 3000);
        })
        .catch(error => {
            dispatch(LoginFailed(error.response.data));
        });
    };

}
