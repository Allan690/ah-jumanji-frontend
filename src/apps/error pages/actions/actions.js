import * as types from './actionTypes';

export default function registerNewError(message, status) {
  return {
    type: types.NEW_ERROR_RAISED,
    hasError:true,
    message,
    status
  };
}
