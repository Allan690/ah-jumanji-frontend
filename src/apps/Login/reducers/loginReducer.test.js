import loginUserReducer from './loginReducer';
import * as types from '../actions/actionTypes';

describe('login reducer', () => {

  it('should return the initial state', () => {
    expect(
      loginUserReducer(undefined, {})).toEqual({});
  });

  it('should handle LOGIN_SUCCESS', () => {
    const action = {
      type: types.LOGIN_SUCCESS,
      user: {
        email: 'beja.emmanuel@gmail.com',
        password: '12345678'
      }
    };

    const expectedState = {
     email: 'beja.emmanuel@gmail.com',
     password: '12345678'
    };

    expect( loginUserReducer( { }, action )).toEqual(expectedState );
  });

  it('should handle LOGIN_FAIL', () => {
    const action = {
      type: types.LOGIN_FAILED
    };
    expect(loginUserReducer({}, action)).toEqual({});
  });

});
