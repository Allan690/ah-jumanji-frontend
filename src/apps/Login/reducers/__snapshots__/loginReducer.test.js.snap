// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login reducer should handle LOGIN_SUCCESS 1`] = `
Object {
  "email": "beja.emmanuel@gmail.com",
  "password": "12345678",
}
`;
